util/dvsim/dvsim.py

```diff
@@ -150,6 +153,29 @@ def get_proj_root():
     return (proj_root)
 
 
+def resolve_proj_root(args):
+    '''Update proj_root based on how DVSim is invoked.
+
+    If --remote env var is set, a location in the scratch area is chosen as the
+    new proj_root. The entire repo is copied over to this location. Else, the
+    proj_root is discovered using get_proj_root() method, unless the user
+    overrides it on the command line.
+
+    This function returns the updated proj_root path.
+    '''
+    proj_root = args.proj_root or get_proj_root()
```

Comment:
Nit: Since `get_proj_root` is only used here, would it make sense to fold it in to this function? Otherwise, maybe we need new names: at the moment, they sound awfully similar.

Reply:
I will leave it as it is, since we are going to have to revisit this in the near future anyway.